python/ray/serve/tests/test_fastapi.py

```diff
@@ -612,6 +612,29 @@ def __del__(self):
     ray.get(del_signal.wait.remote(), timeout=20)
 
 
+def test_fastapi_shutdown_hook_async(serve_instance):
+    # https://github.com/ray-project/ray/issues/41261
+    shutdown_signal = SignalActor.remote()
+    del_signal = SignalActor.remote()
```

Comment:
Nice usage of signals to avoid flaky tests @GeneDer

Reply:
hahaha, this came from copying the test above lol

Reply:
![image](https://github.com/ray-project/ray/assets/9871461/2ff0ed43-b2b3-4e91-adb4-e0b0cdd635b3)